ui/urls.py

```diff
@@ -11,9 +11,11 @@
     url(r'^$', views.Index.as_view(), name='index'),
     url(r'^register/$', views.register, name='register'),
     url(r'^login/$', views.ui_login, name='login'),
-    url(r'^collections/$', views.CollectionList.as_view(), name='collection-list'),
-    url(r'^collections/(?P<collection_key>[0-9a-f]+)/$', views.CollectionDetail.as_view(), name='collection-detail'),
-    url(r'^collections/(?P<collection_key>[0-9a-f]+)/upload/$', views.Upload.as_view(), name='upload'),
+
+    url(r'^collections/', views.CollectionReactView.as_view(), name='collection-react-view'),
```

Comment:
Is it standard when using a React frontend use the same view for both an individual object and a list of objects? This confused me at first, until I saw the React code that determines what to show based on the URL.  It seems less intuitive to me but it works so I'm okay with it.

Reply:
Using React, it's pretty standard to have an entire web application using the same view! If you look at Micromasters (browser-accessible URLs [here](https://github.com/mitodl/micromasters/blob/master/ui/urls.py)), there is just one view (as far as the backend is concerned) for both the learners page and the dashboard page, but those two views represent almost the entire UI for the app.